### PR TITLE
Add job card thumbnails

### LINKED ISSUE #102: Job Card Thumbnails

### PR DESCRIPTION
**Pull request by mas393**
With this PR:
 - When the STL file is loaded in the model viewer, the model is centred and normalized to fit in the viewer
 - After a file upload, a custom thumbnail is taken of the model view
 - When the new job is uploaded, the thumbnail is added to the backend storage 
 - When jobs are fetched for the browse jobs page, or for a user's jobs or orders page, the thumbnail is also fetched from the backend, and displayed w/ the job card.

Closes #102

### CODE REVIEW

**Review by DanielVette — APPROVED** [review]
👍

File: src/components/jobCard.js [hunk 1]
```diff
@@ -21,7 +22,7 @@ export default function JobCard({
       <div className="md:flex-shrink-0">
         <img
           className="h-48 w-full object-cover md:w-48"
-          src={boat}
+          src={(img)? img: boat}
```
Comment (Jasper-Nelligan) [inline src/components/jobCard.js:25]: I think we should replace the boat with a different image if the thumbnail is missing, possibly this one?
no_image

File: src/components/jobCardList.js [hunk 2]
```diff
@@ -14,6 +15,7 @@ export default function JobCardList({ jobs, filters, selectedJob, onSelectJob, o
             isSelected={selectedJob===job}
             onSelectJob={onSelectJob}
             onUnselectJob={onUnselectJob}
+            img={(job.thumbnail)? job.thumbnail: boat}
```
Comment (Jasper-Nelligan) [inline src/components/jobCardList.js:18]: Isn't this logic already handled in `jobCard`? Currently if we pass an empty job thumbnail, then jobCard should handle it.

File: src/routes/browse.js [hunk 3]
```diff
@@ -181,7 +196,7 @@ export default function Browse() {
             <div style={{ display: 'flex', alignItems: 'center' }}>
               <img
                 className="h-96 w-full object-cover md:w-96"
-                src={boat}
+                src={(selectedJob.thumbnail)? selectedJob.thumbnail: boat}
```
Comment (Jasper-Nelligan) [inline src/routes/browse.js:199]: Also here

File: src/routes/create.js [hunk 4]
```diff
@@ -30,63 +32,75 @@ export default function Create() {
...
+    const thumbnailResource = await fetch(thumbnail);
+    const thumbnailBlob = await thumbnailResource.blob();
+    uploadBytes(storageRef, thumbnailBlob);
+  };
+
+  const uploadStl = async (stlFile, id) => {
+    var storageRef = ref(firebaseStorage, `print-files/${id}`);
+    uploadBytes(storageRef, stlFile);
+  }
+
```
Comment (Jasper-Nelligan) [inline src/routes/create.js:46]: Can these methods be moved to backend.js?

File: src/routes/upload.js [hunk 5]
```diff
@@ -20,16 +20,20 @@ export default function Upload({ printJob, updateFile }) {
...
-  spotlight.position.set(40, 40, 40);
-  spotlight.intensity = 3000;
+  const light = new THREE.AmbientLight(0xffffff, 0.1);
+  const spotlight = new THREE.SpotLight(0xffffff);
+  spotlight.position.set(3, 3, 3);
+  spotlight.intensity = 35;
   scene.add(light);
   scene.add(spotlight);
+  //var helper = new THREE.SpotLightHelper(spotlight);
+  //scene.add(helper)
```
Comment (Jasper-Nelligan) [inline src/routes/upload.js:30]: Do we still need the commented out code here?

File: src/routes/configure.js [hunk 6]
```diff
@@ -137,6 +138,9 @@ export default function Configure({ printJob, changePrintJob }) {
 
   return (
     <div>
+      <div>
+        <JobCard job={printJob} onSelectJob={()=>{}} img={printJob.thumbnail}/>
+      </div>
```
Comment (Jasper-Nelligan) [inline src/routes/configure.js:143]: I'm wondering, now that we have the job card showing on the configuration page, would it be possible for the thumbnail to change colors when the user selects a different color of material?

Another thing about adding the job card to here is that when you update material, quantity etc the job card doesn't update which I can see as being confusing the user. I think we should just have the thumbnail on display here

**Pull request merged**